src/tools/kata-ctl/src/check.rs

```diff
@@ -197,6 +212,81 @@ pub fn check_official_releases() -> Result<()> {
     Ok(())
 }
 
+#[cfg(any(target_arch = "x86_64"))]
+pub fn check_kernel_module_loaded(module: &str, parameter: &str) -> Result<String, String> {
+    const MODPROBE_PARAMETERS_DRY_RUN: &str = "--dry-run";
+    const MODPROBE_PARAMETERS_FIRST_TIME: &str = "--first-time";
+    const MODULES_PATH: &str = "/sys/module";
+
+    // Partial check w/ modinfo
+    let output_modinfo = Command::new(MODINFO_PATH)
+        .arg(module)
+        .output()
+        .expect("failed to execute process");
+
+    let status_modinfo = output_modinfo.status;
+
+    // Partial check w/ modprobe
+    let output_modprobe = Command::new(MODPROBE_PATH)
```

Comment:
why are you doing a modprobe here? Looks like the function not only checks but loads the module as well. Can you change the function name to reflect this ? Something like check_and _load_kernel_module.
cc @jodh-intel

Reply:
Also, if the modinfo succeeds, do you need to perform a modprobe?

Reply:
@amshinde:

This function receives two arguments:

- a module name
- a module parameter name

The function determines two things:
- if the module name passed exists
- if the module is currently loaded

In case the module exist and is already loaded, the function returns the value of the parameter passed as argument.

1) why are you doing a modprobe here? 

I've passed two arguments to modprobe:
- --dry-run
- --first-time

`--first-time ` causes modprobe to fail if the module passed is already loaded, and the `--dry-run` flag causes `modprobe` to avoid loading it or removing [1], this is because the function should not load or unload any module passed.

2) modinfo is used to determine if a module exist.

The combination of `modinfo` and `modprobe` are as follows:

`modinfo $module >/dev/null 2>/dev/null && ! modprobe --dry-run --first-time $module`

[1]: https://man7.org/linux/man-pages/man8/modprobe.8.html

Reply:
Ok, thanks for the explanation. Would be helpful to summarise a bit in the comments.